### DISCUSSION TITLE
Is it possible to get the total number of pages when a table's autoPage is true?

### DISCUSSION THREAD

**Discussion by amotzenbecker** [gitbrent/PptxGenJS#1149] [Q&A] [answered]
First off, thanks for this amazing tool. It's been a huge time saver.

I am adding multiple tables to my current slide with `autoPage: true`. This works great. I'm trying to keep track of the current page number for the start of each table. This way, I can reference and link to the slide numbers in an index at the end of the document.

Is there any way to know how many pages/slides `autoPage` has added after creating the table?

I see it was [removed here, noting that developers can keep track of the number of slides added](https://github.com/gitbrent/PptxGenJS/issues/1050#issuecomment-1050415092). I can certainly do this with `addSlide` but I'm not clear on how to do that with tables and autoPage. 

Thanks

**Comment by amotzenbecker** [comment, marked as answer]
I just reviewed the source code and do see that this is available to the slide object as a property: `_slideNum`, e.g.:

```
const currentSlide = pptx.addSlide();
const slideNum = currentSlide._slideNum;
```

**Comment by gitbrent** [comment]
What's the purpose behind knowing the number of slides?

**Reply by kitsonk** [reply]
For me, in complex slide generation scenarios keeping track of the number of slides, especially with some auto paged tables, to be able to update a table of contents slide towards the front of the document is complicated without the information as part of the public API.
